connect/api/src/test/java/org/apache/kafka/connect/data/ConnectSchemaTest.java

```diff
@@ -330,4 +330,127 @@ public void testEmptyStruct() {
+                "Invalid Java object for schema with type STRING: class java.lang.Boolean for field: \"value\"");
+        assertInvalidValueForSchema(fieldName, optionalStrings, Collections.singletonMap(true, "value"),
+                "Invalid Java object for schema with type STRING: class java.lang.Boolean for field: \"key\"");
+
+        // Required element schema
+        Schema requiredStrings = SchemaBuilder.map(Schema.STRING_SCHEMA, Schema.STRING_SCHEMA);
+        ConnectSchema.validateValue(fieldName, requiredStrings, Collections.emptyMap());
+        ConnectSchema.validateValue(fieldName, requiredStrings, Collections.singletonMap("key", "value"));
+        assertInvalidValueForSchema(fieldName, requiredStrings, Collections.singletonMap("key", null),
+                "Invalid value: null used for required field: \"value\", schema type: STRING");
+        assertInvalidValueForSchema(fieldName, requiredStrings, Collections.singletonMap(null, "value"),
+                "Invalid value: null used for required field: \"key\", schema type: STRING");
+        assertInvalidValueForSchema(fieldName, requiredStrings, Collections.singletonMap(null, null),
+                "Invalid value: null used for required field: \"key\", schema type: STRING");
+        assertInvalidValueForSchema(fieldName, requiredStrings, Collections.singletonMap("key", true),
+                "Invalid Java object for schema with type STRING: class java.lang.Boolean for field: \"value\"");
+        assertInvalidValueForSchema(fieldName, requiredStrings, Collections.singletonMap(true, "value"),
+                "Invalid Java object for schema with type STRING: class java.lang.Boolean for field: \"key\"");
+
+        // Null element schema
+        Schema nullElements = SchemaBuilder.type(Schema.Type.MAP);
+        assertInvalidValueForSchema(fieldName, nullElements, Collections.emptyMap(),
+                "No schema defined for keys of field: \"field\"");
+        assertInvalidValueForSchema(fieldName, nullElements, Collections.singletonMap("key", "value"),
+                "No schema defined for keys of field: \"field\"");
+        assertInvalidValueForSchema(fieldName, nullElements, Collections.singletonMap("key", null),
+                "No schema defined for keys of field: \"field\"");
+        assertInvalidValueForSchema(fieldName, nullElements, Collections.singletonMap(null, "value"),
+                "No schema defined for keys of field: \"field\"");
+        assertInvalidValueForSchema(fieldName, nullElements, Collections.singletonMap(null, null),
+                "No schema defined for keys of field: \"field\"");
+        assertInvalidValueForSchema(fieldName, nullElements, Collections.singletonMap("key", true),
+                "No schema defined for keys of field: \"field\"");
+        assertInvalidValueForSchema(fieldName, nullElements, Collections.singletonMap(true, "value"),
+                "No schema defined for keys of field: \"field\"");
```

Comment:
If it's not too much work, it could also be nice to exercise the code path where there's a key schema defined but no value schema. But only if it's a super easy addition, otherwise not worth the work.